### PR TITLE
Extract URL

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [ס] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39613

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: Can click on issue above to see BC 
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by tomneeman151293** [comment]
@dantavori Only need to add a TPB and update release notes, but apart from that PR seems ready from my side.

**Comment by tomneeman151293** [comment]
@melamedbn @dantavori Ready for review!

File: Packs/CommonScripts/TestPlaybooks/FormatURL-Test.yml [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+        "height": 2310,
+        "width": 380,
+        "x": 50,
+        "y": 50
+      }
+    }
+  }
+inputs: []
+outputs: []
+fromversion: 5.5.0
```
Comment (dantavori) [inline Packs/CommonScripts/TestPlaybooks/FormatURL-Test.yml:540]:
```suggestion
fromversion: 6.1.0
```

File: Packs/CommonScripts/Scripts/FormatURL/FormatURL.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  type: String
+script: '-'
+system: false
+tags:
+- indicator-format
+timeout: '0'
+type: python
+subtype: python3
+dockerimage: demisto/python3:3.9.8.24399
+fromversion: 5.5.0
```
Comment (dantavori) [inline Packs/CommonScripts/Scripts/FormatURL/FormatURL.yml:26]:
```suggestion
fromversion: 6.1.0
```

File: Packs/CommonScripts/Scripts/FormatURL/FormatURL.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+args:
+- default: false
+  description: URL inputs. Comma separated list.
```
Comment (dantavori) [inline Packs/CommonScripts/Scripts/FormatURL/FormatURL.yml:3]:
```suggestion
  description: A comma-separated list of URL inputs.
```

File: Packs/CommonScripts/Scripts/FormatURL/FormatURL.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  isArray: true
+  name: input
+  required: true
+  secret: false
+comment: "Formats URL to indicator: extracts redirect URL from Proof Point or ATP URLs. Strips and unquotes and unescapes URLs"
```
Comment (dantavori) [inline Packs/CommonScripts/Scripts/FormatURL/FormatURL.yml:8]:
```suggestion
comment: "Strips, unquotes and unescapes URLs. If the URL is a Proofpoint or ATP URL, extracts its redirect URL."
```

File: Packs/CommonScripts/ReleaseNotes/1_6_15.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#### Scripts
+##### New: FormatURL
+- Formats URL to indicator: extracts redirect URL from Proof Point or ATP URLs. Strips and unquotes and unescapes URLs (Available from Cortex XSOAR 5.5.0).
```
Comment (dantavori) [inline Packs/CommonScripts/ReleaseNotes/1_6_15.md:3]:
```suggestion
- Strips, unquotes and unescapes URLs. If the URL is a Proofpoint or ATP URL, extracts its redirect URL. (Available from Cortex XSOAR 6.1.0).
```

File: Packs/CommonTypes/ReleaseNotes/3_1_31.json [hunk 6]
```diff
@@ -0,0 +1 @@
+{"breakingChanges": true, "breakingChangesNotes": null}
```
Comment (dantavori) [inline Packs/CommonTypes/ReleaseNotes/3_1_31.json:1]: why are we not adding the breakingChangesNotes?

File: Packs/CommonTypes/ReleaseNotes/3_1_31.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+#### Indicator Types
+- **urlRep**: Changed the formatting script default to **FormatURL** script.
+- **Breaking Change**: The URL indicator regex will no longer extract domains, e.g `www.test.com` will not be extracted as URL.
```
Comment (dantavori) [inline Packs/CommonTypes/ReleaseNotes/3_1_31.md:3]:
```suggestion
- **Breaking Change**: The URL indicator regex will no longer extract domains, E.g., `www.test.com` will not be extracted as a URL.
```

File: Packs/CommonScripts/Scripts/FormatURL/README.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
+Formats URL to indicator: extracts redirect URL from Proof Point or ATP URLs. Strips and unquotes and unescapes URLs
```
Comment (dantavori) [inline Packs/CommonScripts/Scripts/FormatURL/README.md:1]: please regenerate

**Review by dantavori — CHANGES_REQUESTED** [review]
amazing as always, see my small notes

File: Packs/CommonTypes/ReleaseNotes/3_1_31.json (hunk 6 above)
Comment (tomneeman151293) [reply Packs/CommonTypes/ReleaseNotes/3_1_31.json:1]: If it is null, by the BC design it will just show the whole RN file so i didnt add specific text

**Review by dantavori — APPROVED** [review]

**Pull request merged**